controllers/cloud.redhat.com/namespacereservation_controller.go

```diff
@@ -106,14 +106,19 @@ func (r *NamespaceReservationReconciler) Reconcile(ctx context.Context, req ctrl
 			return ctrl.Result{}, err
 		}
 		if r.Poller.namespaceIsExpired(expirationTS) {
-			err = DeletePrometheusOperator(ctx, r.Client, res.Status.Namespace)
-			if k8serr.IsNotFound(err) {
-				r.Log.Error(err, fmt.Sprintf("the prometheus operator prometheus.%s does not exist.", res.Status.Namespace))
+			removed, err := CheckForSubscriptionPrometheusOperator(ctx, r.Client, res.Status.Namespace)
+			if err != nil {
+				r.Log.Error(err, fmt.Sprintf("prometheus operator subscription for namespace %s still exists.", res.Status.Namespace))
 				return ctrl.Result{Requeue: true}, err
-			} else if err != nil {
-				r.Log.Error(err, fmt.Sprintf("Error deleting prometheus.%s", res.Status.Namespace))
-			} else {
-				r.Log.Info("Successfully deleted", "prometheus-operator", fmt.Sprintf("prometheus.%s", res.Status.Namespace))
+			} else if removed {
+				r.Log.Info("Subscription for prometheus operator was deleted", "prometheus-operator subscription", fmt.Sprint(res.Status.Namespace))
+			}
+
+			removed, err = DeletePrometheusOperator(ctx, r.Client, res.Status.Namespace)
+			if err != nil {
+				r.Log.Error(err, fmt.Sprintf("prometheus operator subscription for namespace %s still exists.", res.Status.Namespace))
```

Comment:
Should we requeue here at this point if deleting the prometheus operator returned an err?